mongo/src/main/java/com/redhat/lightblue/mongo/crud/Translator.java

```diff
@@ -766,30 +772,34 @@ public static void populateDocHiddenFields(DBObject doc, EntityMetadata md) thro
         populateDocHiddenFields(doc, fieldMap);
     }
 
+    /**
+     *
+     * @param doc
+     * @param fieldMap <field, hiddenCounterpart>
+     * @throws IOException
+     */
     public static void populateDocHiddenFields(DBObject doc, Map<String, String> fieldMap) throws IOException {
         for (String index : fieldMap.keySet()) {
             int arrIndex = index.indexOf("*");
             if (arrIndex > -1) {
                 // recurse if we have more arrays in the path
                 populateHiddenArrayField(doc, index, fieldMap.get(index));
             } else {
-                Object dbObject = null;
-                try {
-                    dbObject = getDBObject(doc, new Path(index));
-                } catch (Exception e) {
-                    LOGGER.debug("Error when populating hidden field {} with value from canonical field {}\n"
-                            + "Document being populated: \n{}", fieldMap.get(index), index, doc);
-                    throw e;
+                DBObject currentDbo = doc;
+                Path path = new Path(index);
+                for(int i = 0; i < path.numSegments() - 1; i++){
+                    // recurse down the obj tree and
+                    currentDbo = (DBObject) currentDbo.get(path.head(i));
                 }
-                if (dbObject != null) {
-                    ObjectNode arrNode = JsonNodeFactory.instance.objectNode();
-                    LOGGER.debug("Adding field '" + fieldMap.get(index) + "' to document with value " + dbObject.toString().toUpperCase());
-                    JsonDoc.modify(arrNode, new Path(fieldMap.get(index)), JsonNodeFactory.instance.textNode(dbObject.toString().toUpperCase()), true);
-                    ObjectMapper mapper = new ObjectMapper();
-                    JsonNode merged = merge(mapper.readTree(doc.toString()), mapper.readTree(arrNode.toString()));
-                    DBObject obj = (DBObject) JSON.parse(merged.toString());
-                    ((BasicDBObject) doc).clear();
-                    ((BasicDBObject) doc).putAll(obj);
+                // given the last basic object, populate its hidden field
+                String val = (String) currentDbo.get(path.getLast());
+                if (val != null) {
+                    DBObject hidden = (DBObject) currentDbo.get(HIDDEN_SUB_PATH.toString());
+                    if (hidden == null) {
+                        currentDbo.put(HIDDEN_SUB_PATH.toString(), new BasicDBObject(path.getLast(), val.toUpperCase()));
+                    } else {
+                        hidden.put(path.getLast(), val.toUpperCase());
+                    }
                 }
```

Comment:
Good catch.  We didn't have that happening before and this logic makes it quite a bit easier to implement.  I think there's another issue out there that we can close.